exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryResources.java

```diff
@@ -110,6 +118,48 @@ public Viewable submitQuery(@FormParam("query") String query,
     }
   }
 
+  @Nonnull
+  private Map<String, Object> readOptionsFromForm(Form form) {
+    Map<String, Object> options = new HashMap<String, Object>();
+    SessionOptionManager sessionOptionManager = webUserConnection.getSession().getOptions();
+    for (Map.Entry<String, List<String>> pair : form.asMap().entrySet()) {
+      String name = pair.getKey();
+      OptionDefinition definition = sessionOptionManager.getOptionDefinition(name);
+      if (definition != null) {
+        String valueStr = pair.getValue().get(0);
+        if (!valueStr.isEmpty()) {
+          try {
+            Object value = null;
+            switch (definition.getValidator().getKind()) {
+              case BOOLEAN:
```

Comment:
Nit: `value = Boolean.parseBoolean(valueStr);` (Or `valueOf()`, which seems to be the new style.)

Reply:
I considered it, but I wanted to validate that the input was either `true` or `false`, so I wrote my own algorithm.  `parseBoolean` will just return `false` if the input isn't `"true"` but otherwise it accepts anything.